frontend/src/components/password/PasswordInput.tsx

```diff
@@ -19,6 +19,10 @@ export default function PasswordInput({
   const [password, setPassword] = React.useState(value);
   const [passwordVisible, setPasswordVisible] = React.useState(false);
 
+  React.useEffect(() => {
```

Comment:
This seems like a hack to me, it shouldn't be needed if indeed `unlockPassword` is changed because it's passed in which will cause a re-render

Reply:
Can you double check if your assumptions are right about the root cause?

Reply:
Yup, this is the root cause. Prop change doesn't necessarily mean a re-render right? Since the state has been initialized here and changing the `value` prop doesn't change the `password` state variable

Reply:
But I agree this looks like a hack, I'll try to remove the internal state because even the other useEffect looks bad in this component (can't we just call onChange directly?)

Reply:
Done